sycl/test/filter_selector/select.cpp

```diff
@@ -0,0 +1,133 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t1.out
+// RUN: %t1.out
+
+//==------------------- select.cpp - string_selector test ------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+
+using namespace cl::sycl;
+using namespace cl::sycl::ONEAPI;
+
+int main() {
+  std::vector<device> CPUs;
+  std::vector<device> GPUs;
+  std::vector<device> Devs;
+  device host;
+
+  CPUs = device::get_devices(info::device_type::cpu);
+  GPUs = device::get_devices(info::device_type::gpu);
+  Devs = device::get_devices();
+
+  std::cout << "# CPU Devices found: " << CPUs.size() << std::endl;
+  std::cout << "# GPU Devices found: " << GPUs.size() << std::endl;
+  std::cout << "# Devices found: " << Devs.size() << std::endl;
+
+  bool HasLevelZero = false;
+  bool HasOpenCL = false;
+
+  auto Platforms = platform::get_platforms();
+  for (auto Platform : Platforms) {
+    if (!Platform.is_host()) {
+      auto Backend = Platform.get_backend();
+      if (Backend == backend::level_zero) {
+        HasLevelZero = true;
+      } else if (Backend == backend::opencl) {
```

Comment:
CUDA BE is missed

Reply:
I wasn't sure CUDA was enabled on the test machine - is it?

Reply:
there is separate machine with CUDA RT installed to test the configuration.

Reply:
Ok